src/ecdsa/test_ecdh.py

```diff
@@ -322,10 +319,18 @@ def test_ecdh_with_openssl(vcurve):
     run_openssl("pkeyutl -derive -inkey t/privkey1.pem -peerkey t/pubkey2.pem -out t/secret1")
     run_openssl("pkeyutl -derive -inkey t/privkey2.pem -peerkey t/pubkey1.pem -out t/secret2")
 
-    with open("t/secret1", "rb") as e:
-        ssl_secret1 = e.read()
-    with open("t/secret1", "rb") as e:
-        ssl_secret2 = e.read()
+    try:
+        with open("t/secret1", "rb") as e:
+            ssl_secret1 = e.read()
+        with open("t/secret1", "rb") as e:
+            ssl_secret2 = e.read()
+    except FileNotFoundError:
+        pytest.skip("system openssl does not support `pkeyutl -derive`")
```

Comment:
I'm pretty sure that if openssl doesn't support -derive we won't get this far: the run_openssl will raise subprocess.SubprocessError exception because the exit code of command will be non-zero, won't it?

looking for `-derive` in `pkeyutl -help` output may be a better idea...